BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -21,9 +22,17 @@ public class Transmission
 
         private static readonly TimeSpan DefaultTimeout = TimeSpan.FromSeconds(100);
         private static HttpClient client = new HttpClient() { Timeout = System.Threading.Timeout.InfiniteTimeSpan };
-                
+        private static long flushAsyncCounter = 0;
+
         private int isSending;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Transmission"/> class. This overload seperates telemetryitems serialization from object construction. 
+        /// </summary>
+        public Transmission()
```

Comment:
If the goal is to allow other packages (e.g. telemetry channel implementations) to access this ctor, a safer approach would be https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute.

Reply:
@cijothomas, We had offline discussion about this topic and decided not to use internals visible. Introducing internals visible will limit our public API changes, should we reconsider  `InternalsVisibleToAttribute`?

Reply:
We can make this new fields internal, and use InternalsVisible approach now given that its intended to be accessible from ServerTelemetryChannel now.
If a need arises in the future, we can make it public, so other channels may leverage this.

Reply:
Can we make the InternalsVisible change as a separate PR, please?

Reply:
If its part of this PR, it'll simply the .public API file in this PR a lot.

Reply:
@TimothyMothra I agree with you. This has to be done separately. I will send a new PR and get that merged before this one.